extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/cache/CacheInvalidateAllInterceptor.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.infinispan.client.runtime.cache;
+
+import java.util.function.Function;
+
+import javax.annotation.Priority;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.commons.CacheException;
+import org.jboss.logging.Logger;
+
+import io.quarkus.infinispan.client.CacheInvalidateAll;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+
+@CacheInvalidateAll(cacheName = "") // The `cacheName` attribute is @Nonbinding.
+@Interceptor
+@Priority(CacheInterceptor.BASE_PRIORITY)
+public class CacheInvalidateAllInterceptor extends CacheInterceptor {
+
+    private static final Logger LOGGER = Logger.getLogger(CacheInvalidateAllInterceptor.class);
+    private static final String INTERCEPTOR_BINDINGS_ERROR_MSG = "The Quarkus Infinispan Client extension is not working properly (CacheInvalidateAll interceptor bindings retrieval failed), please create a GitHub issue in the Quarkus repository to help the maintainers fix this bug";
+
+    @AroundInvoke
+    public Object intercept(InvocationContext invocationContext) throws Exception {
+        CacheInterceptionContext<CacheInvalidateAll> interceptionContext = getInterceptionContext(invocationContext,
+                CacheInvalidateAll.class);
+        if (interceptionContext.getInterceptorBindings().isEmpty()) {
+            // This should never happen.
+            LOGGER.warn(INTERCEPTOR_BINDINGS_ERROR_MSG);
+            return invocationContext.proceed();
+        }
+        ReturnType returnType = determineReturnType(invocationContext.getMethod().getReturnType());
+        if (returnType == ReturnType.NonAsync) {
+            return invalidateAllBlocking(invocationContext, interceptionContext);
+        }
+        return invalidateAllNonBlocking(invocationContext, interceptionContext);
+    }
+
+    private Object invalidateAllNonBlocking(InvocationContext invocationContext,
+            CacheInterceptionContext<CacheInvalidateAll> interceptionContext) {
+        LOGGER.trace("Invalidating all cache entries in a non-blocking way");
+        return Multi.createFrom().iterable(interceptionContext.getInterceptorBindings())
+                .onItem().transformToUniAndMerge(new Function<CacheInvalidateAll, Uni<? extends Void>>() {
+                    @Override
+                    public Uni<Void> apply(CacheInvalidateAll binding) {
+                        return invalidateAll(binding);
+                    }
+                })
+                .onItem().ignoreAsUni()
+                .onItem().transformToUni(new Function<Object, Uni<?>>() {
+                    @Override
+                    public Uni<?> apply(Object ignored) {
+                        try {
+                            return (Uni<Object>) invocationContext.proceed();
```

Comment:
Some details from #24894 are missing in this class (at this line but also at other lines in the class).

Reply:
This comment and two other ones were marked as resolved but I don't see any changes. Is the fix about to be committed?

Reply:
yes, I have ot locally ;) is for me to mark it as done